test/functional/mempool_datacarrier.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test datacarrier functionality"""
+from test_framework.messages import (
+    CTxOut,
+    CTransaction,
+    MAX_OP_RETURN_RELAY,
+)
+from test_framework.script import (
+    CScript,
+    OP_RETURN,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_node import TestNode
+from test_framework.util import (
+    assert_raises_rpc_error,
+    random_bytes,
+)
+from test_framework.wallet import MiniWallet
+
+
+class DataCarrierTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.extra_args = [
+            [],
+            ["-datacarrier=false"],
+            ["-datacarrier=true", f"-datacarriersize={MAX_OP_RETURN_RELAY - 1}"]
```

Comment:
This parses as `=1` (The test fails regardless of the second option)

Probably a good time to make this an init error finally.

Reply:
FYI for anyone else confused about this, see the docs of [`InterpretBool()`](https://github.com/bitcoin/bitcoin/blob/a6fc293c0a1f27ba1e573bfa16fd76d5f58988b2/src/util/system.cpp#L163-L167):

> non-numeric string values like "foo", return 0

so `-datacarrier=false` and `-datacarrier=true` all resolve to `datacarrier=0`. 

As such, should be:

```py
            ["-datacarrier=0"],
            ["-datacarrier=1", f"-datacarriersize={MAX_OP_RETURN_RELAY - 1}"]
```

Good catch MarcoFalke!

Reply:
Interesting! I applied the requested change and added a new test that checks that  "-datacarrier=1"  works.

Reply:
See also https://github.com/bitcoin/bitcoin/pull/16545